### PR TITLE
feat(compute/rust): Handle Cargo config filename for Rust >=1.78.0

### PR DESCRIPTION
**Pull request by kpfleming**
The preferred Cargo configuration filename has been 'config.toml' since Rust 1.39.0 was released, but 'config' was supported for backwards compatibility (although deprecated).
    
In Rust 1.78.0, the backwards compatibility was removed, so any file with the old name will be ignored. Since Compute package builds require the content in that file, ignoring it results in a build failure.
    
This patch adds checking for this situation. If the old file name is found, and the Rust version is <1.78.0, a warning is issued to let the user know that they should rename the file before upgrading to 1.78.0 or later.  If the old file name is found, and the Rust version is >=1.78.0, an error is emitted and the build process stopped, as it will not succeed.

Example output (with old file name and Rust 1.74.0):
```
kpfleming@kpfleming:~/src/fastly/compute-starter-kit-rust-default$ ~/src/fastly/cli/fastly compute build
✓ Verifying fastly.toml
✓ Identifying package name
✓ Identifying toolchain
WARNING: The Cargo configuration file name is .cargo/config
WARNING: The file should be renamed to .cargo/config.toml to be compatible with Rust 1.78.0 or later

✓ Running [scripts.build]
✓ Creating package archive
```
Example output (with old file name and Rust 1.78.0):
```
kpfleming@kpfleming:~/src/fastly/compute-starter-kit-rust-default$ ~/src/fastly/cli/fastly compute build
✓ Verifying fastly.toml
✓ Identifying package name
✓ Identifying toolchain
WARNING: The Cargo configuration file name is .cargo/config

ERROR: The build cannot proceed with Rust version '1.78.0' as the file must be named .cargo/config.toml.

If you believe this error is the result of a bug, please file an issue: https://github.com/fastly/cli/issues/new?labels=bug&template=bug_report.md
```

### CODE REVIEW

File: pkg/commands/compute/language_rust.go [hunk 1]
```diff
@@ -312,32 +319,57 @@ func (r *Rust) toolchainConstraint() {
...
+		return nil
+	}
+
+	text.Warning(r.output, "The Cargo configuration file name is .cargo/config")
+
+	c, _ := semver.NewConstraint(">=1.78.0")
+
+	if c.Check(rustVersion) {
+		return fmt.Errorf("The build cannot proceed with Rust version '%s' as the file must be named .cargo/config.toml", rustVersion)
+	} else {
```
Comment (Integralist) [inline pkg/commands/compute/language_rust.go:369]: As suggested by the GitHub linter...

Screenshot 2024-05-08 at 06 42 25

File: pkg/commands/compute/language_rust.go [hunk 2]
```diff
@@ -312,32 +319,57 @@ func (r *Rust) toolchainConstraint() {
...
+	if err != nil {
+		r.errlog.Add(err)
+		return fmt.Errorf("getting current working directory: %w", err)
+	}
+
+	if !filesystem.FileExists(filepath.Join(dir, ".cargo", "config")) {
+		return nil
+	}
+
+	text.Warning(r.output, "The Cargo configuration file name is .cargo/config")
```
Comment (Integralist) [inline pkg/commands/compute/language_rust.go:363]: Let's get some breathing space between the WARNING and the preceding line (as I can see from your example output in the PR description that they're butted up against each other)
```suggestion
	text.Warning(r.output, "\nThe Cargo configuration file name is .cargo/config")
```

**Review by Integralist — COMMENTED** [review]
This isn't the end of the world but I would prefer not to have two WARNING lines next to each other when they both relate to the same warning.

I can see why that has happened by looking at the code and the fact you're trying to swap out the second warning for an error (depending on the situation) but it just looks a bit weird.

File: pkg/commands/compute/language_rust.go (hunk 2 above)
Comment (Integralist) [inline pkg/commands/compute/language_rust.go:363]: To fix the issue of a double WARNING message we should change the flow to...

```go
warnMsg := "\nThe Cargo configuration file name is .cargo/config"

c, _ := semver.NewConstraint(">=1.78.0")

if c.Check(rustVersion) {
    text.Warning(r.output, warnMsg)
    return fmt.Errorf("The build cannot proceed with Rust version '%s' as the file must be named .cargo/config.toml", rustVersion)
}

text.Warning(r.output, warnMsg + ". The file should be renamed to .cargo/config.toml to be compatible with Rust 1.78.0 or later\n\n")
return nil
```

Comment (kpfleming) [reply pkg/commands/compute/language_rust.go:363]: This makes sense, will do it now.

**Comment by kpfleming** [comment]
New output:

```
kpfleming@kpfleming:~/src/fastly/compute-starter-kit-rust-default$ ~/src/fastly/cli/fastly compute build
✓ Verifying fastly.toml
✓ Identifying package name
✓ Identifying toolchain

WARNING: The Cargo configuration file name is .cargo/config

ERROR: The build cannot proceed with Rust version '1.78.0' as the file must be named .cargo/config.toml.

If you believe this error is the result of a bug, please file an issue: https://github.com/fastly/cli/issues/new?labels=bug&template=bug_report.md
```

and

```
kpfleming@kpfleming:~/src/fastly/compute-starter-kit-rust-default$ ~/src/fastly/cli/fastly compute build
✓ Verifying fastly.toml
✓ Identifying package name
✓ Identifying toolchain

WARNING: The Cargo configuration file name is .cargo/config. The file should be renamed to
.cargo/config.toml to be compatible with Rust 1.78.0 or later

✓ Running [scripts.build]
✓ Creating package archive

SUCCESS: Built package (pkg/Default-starter-for-Rust.tar.gz)
```

**Review by Integralist — APPROVED** [review]
🎉

**Pull request merged**